### ISSUE TITLE
loss of model decreased well but has terrible prediction result

### ISSUE THREAD

**Issue by cccc0der** [microsoft/Megatron-DeepSpeed#280]
This is a discussion, not a bug report.

I'm using megatron-deepspeed to train a llama likely 13B model,   the loss decreased well from 10 to 2.1~2.2.

Then I converted checkpoint with huggingface modelling to do predict, but the result is very terrible.

Weird thing is, after some SFT training(use converted model), result of the model looks fine.

My question is, is the bad result of pretrained model caused by the code I used to convert?

**Comment by LiuXTao** [comment]
+1

**Comment by sz128** [comment]
In my case, the tokenization in Megatron-DeepSpeed does not contain a `<s>` token at the start of a document. If I try to generate text by using ```tokenizer(prompt, return_tensors="pt")```, the result is weird.  However, ```tokenizer(prompt, return_tensors="pt", add_special_tokens=False)``` can obtain a good result.  The reason is that the `<s>` token is not trained in pre-training.
